src/Search.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Products from './Products';
+
+class _Search extends React.Component{
+    constructor(){
+        super()
+        this.state =  {
+            searchItems: [],
+            inpu: ''
+        }
+    }
+     startSearch (e) {
+         const searchItems = this.props.products.filter(product => product.name.toLowerCase().includes(e.target.value.toLowerCase()))
+         
+         this.setState({ searchItems, input: e.target.value })
+    }
+
+    render(){
+       const { searchItems, input } = this.state
+
+   return( 
+       <div>
+           <form>
+                <input type="text" name="search" id="search" onKeyUp={e => this.startSearch(e)}/>
+            </form>
+            <div className='ordering' >
+          {searchItems.length === 0 && !input ? <Products />:
+            searchItems.map(product => <div key={product.id} className='border' >
+              {product.name}
+              <br/>
+              ${product.price}
+              <br/>
+              <button onClick={() => this.props.toCreate(product)} >Add to Cart</button>
```

Comment:
need to add the toCreate: (item) => dispatch(addCartItem(item)) to the mapStateToDispatch

Reply:
I just copy Product component code by Mathew to safe time.
and we need it later when fire dispatch(attToCart(item))